sdk/containerregistry/azure-containerregistry/azure/containerregistry/_container_registry_client.py

```
@@ -1037,9 +1036,10 @@ def delete_blob(self, repository: str, tag_or_digest: str, **kwargs) -> None:
             from azure.identity import DefaultAzureCredential
             endpoint = os.environ["CONTAINERREGISTRY_ENDPOINT"]
             client = ContainerRegistryClient(endpoint, DefaultAzureCredential(), audience="my_audience")
-            client.delete_blob("my_repository", "my_tag_or_digest")
+            client.delete_blob("my_repository", "my_digest")
         """
-        if _is_tag(tag_or_digest):
-            tag_or_digest = self._get_digest_from_tag(repository, tag_or_digest)
-
-        self._client.container_registry_blob.delete_blob(repository, tag_or_digest, **kwargs)
+        try:
+            self._client.container_registry_blob.delete_blob(repository, digest, **kwargs)
+        except HttpResponseError as error:
+            if error.status_code == 404:
+                return
```

You need to add the final `raise` or all errors will be suppressed.